es/ae/oracle.js

```diff
@@ -121,22 +118,15 @@ async function getQueryObject (oracleId, queryId) {
  * @return {Promise<Object>} OracleQuery object
  */
 export async function pollForQueryResponse (oracleId, queryId, { attempts = 20, interval = 5000 } = {}) {
-  const emptyResponse = 'or_Xfbg4g=='
-  async function pause (duration) {
-    await new Promise(resolve => setTimeout(resolve, duration))
-  }
-  async function probe (left) {
-    const query = await this.getOracleQuery(oracleId, queryId)
-    if (query.response !== emptyResponse) {
-      return { response: query.response, decode: () => decodeBase64Check(query.response.slice(3)) }
-    }
-    if (left > 0) {
-      await pause(interval)
-      return probe.bind(this)(left - 1)
+  for (let i = 0; i < attempts; i++) {
```

Comment:
are you sure the for correctly awaits the promise here for each iteration?

I'd maybe go for a more high-level approach.

1. use a range array instead of `for i` (e.g. as in https://github.com/aeternity/aepp-governance/blob/master/governance-server/src/util.js#L45)
2. use an async reduce to iterate and eventually break the iteration (maybe something similar to https://github.com/aeternity/aepp-governance/blob/master/governance-server/src/util.js#L58 can be helpful)

Reply:
This could also be combined in a util function, e.g. as `attemptWithPause(() => ...)` passing the function, where e.g. a rejected promise or returned false is the break condition

Reply:
> are you sure the for correctly awaits the promise here for each iteration?

Yes, I think so. Actually, I'm rarely writing a `for` cycle, but I think it is a good place to use it. I prefer this because it uses just plain js.